packages/admin-ui/ui/src/domain/product-categories/modals/add-product-category.tsx

```diff
@@ -45,23 +86,33 @@ function CreateProductCategory(props: CreateProductCategoryProps) {
   const notification = useNotification()
   const queryClient = useQueryClient()
 
-  const [name, setName] = useState("")
-  const [handle, setHandle] = useState("")
-  const [description, setDescription] = useState("")
-  const [isActive, setIsActive] = useState(true)
-  const [isPublic, setIsPublic] = useState(true)
+  const form = useForm<CategoryFormData>({
+    defaultValues: getDefaultCategoryValues(t),
+    mode: "onChange",
+  })
+  const {
+    register,
+    handleSubmit,
+    watch,
+    reset,
+    control,
+    formState: { errors, isDirty, isValid, isSubmitting },
+  } = form
+  const name = watch("name", "")
 
-  const { mutateAsync: createProductCategory } = useAdminCreateProductCategory()
+  const { mutateAsync: createProductCategory } =
+    useAdminCreateProductCategory()
 
-  const onSubmit = async () => {
+  const submit = async (data: CategoryFormData) => {
```

Comment:
```suggestion
  const submit = handleSubmit(async (data) => {
```
Might as well wrap it here, and let the type of data be inferred instead of casting it.